specification/terraform/Microsoft.AzureTerraform.Management/routes.tsp

```diff
@@ -2,52 +2,60 @@ import "@azure-tools/typespec-azure-core";
 import "@typespec/rest";
 import "./models.tsp";
 import "@azure-tools/typespec-azure-resource-manager";
+import "@azure-tools/typespec-client-generator-core";
 
 using TypeSpec.Rest;
 using TypeSpec.Http;
+using Azure.Core;
+using Azure.ClientGenerator.Core;
 using Azure.ResourceManager;
 using Azure.ResourceManager.Foundations;
 using OpenAPI;
 
 namespace Microsoft.AzureTerraform;
 
-#suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-operation" "Cannot use @armResourceOperations decorator here, the auto-generated routes do not match feature requirements"
-#suppress "@azure-tools/typespec-azure-core/no-openapi" "TODO: migrate to LRO concepts DO NOT USE x-ms-long-running-operation-options"
-@doc("Exports the Terraform configuration of the specified resource(s).")
-@route("/subscriptions/{subscriptionId}/providers/Microsoft.AzureTerraform/exportTerraform")
-@post
-@tag("ExportTerraform")
-@extension(
-  "x-ms-long-running-operation-options",
-  {
-    `final-state-via`: "azure-async-operation",
-    `final-state-schema`: "#/definitions/OperationStatus",
-  }
-)
-op exportTerraform(
-  ...ApiVersionParameter,
-  ...SubscriptionIdParameter,
+@doc("Parameters common to all ExportTerraform operations")
+model CommonExportTerraformParameters {
+  ...ApiVersionParameter;
+  ...SubscriptionIdParameter;
+}
+
+#suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-interface-requires-decorator" "Cannot use @armResourceOperations, the implied @autoRoute produces the wrong route for GET /operationStatuses/{operationId}"
+@route("/subscriptions/{subscriptionId}/providers/Microsoft.AzureTerraform")
+interface ExportTerraform {
```

Comment:
so the linter is reporting a warning: OperationId should contain the verb: 'exportterraform' in:'ExportTerraform_Export'. Consider updating the operationId
Location: [Microsoft.AzureTerraform/preview/2023-07-01-preview/export.json#L91](https://github.com/Azure/azure-rest-api-specs/blob/e669bc3b30c40d396f96e90dc60eeccf57ff4d66/specification/terraform/resource-manager/Microsoft.AzureTerraform/preview/2023-07-01-preview/export.json#L91)
I think it makes sense.
Could we change the name of this interface so that its name is a noun?
Usually for resources, it is `VirtualMachines_Create`, which means the name of the interface is `VirtualMachines`.
For your case, maybe it should be `Terraform` or `AzureTerraform`?

Reply:
Fixed in 405a932dae by renaming the Interface. The operationId is now `Terraform_Export`.

Reply:
I have tried all the following operationIds:

- `Terraform_Export`
- `ExportTerraform_ExportTerraform`
- `Terraform_ExportTerraform`
- `ExportTerraform_Export`

And either of these linters fail:

- [operation-id-noun-verb.md](https://github.com/Azure/azure-openapi-validator/blob/main/docs/operation-id-noun-verb.md)
- [post-operation-id-contains-url-verb.md](https://github.com/Azure/azure-openapi-validator/blob/main/docs/post-operation-id-contains-url-verb.md)

I will have to suppress this linter for the PR.